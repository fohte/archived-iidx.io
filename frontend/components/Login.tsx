import { auth } from '../lib/firebaseApp'

const handleLogin = async () => {
  try {
    await auth.signInWithEmailAndPassword(
      window.prompt('email') || '',
      window.prompt('password') || '',
    )
  } catch (e) {
    console.error(e)
  }
}

export default () => <button onClick={handleLogin}>login</button>
