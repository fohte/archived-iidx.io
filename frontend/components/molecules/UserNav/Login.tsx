import { Button } from 'antd'
import Link from 'next/link'
import * as React from 'react'

const Login: React.SFC = () => (
  <Link href="/login" prefetch>
    <Button>Login</Button>
  </Link>
)

export default Login
