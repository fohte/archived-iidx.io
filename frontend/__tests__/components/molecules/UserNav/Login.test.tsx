import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Login from '@app/components/molecules/UserNav/Login'

describe('Login', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Login />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
