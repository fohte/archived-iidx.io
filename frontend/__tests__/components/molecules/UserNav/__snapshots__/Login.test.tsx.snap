// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<button
  className="ant-btn"
  onClick={[Function]}
  type="button"
>
  <span>
    Login
  </span>
</button>
`;
